content/ui-patterns/loading.mdx

```diff
@@ -0,0 +1,402 @@
+
+If the completed process results in navigating to a new page, just rely on the browser's default focus behavior for new pages.
+
+If the process fails and a validation message is shown, move focus to the first focusable element in the error message. See the [form validation guidelines](/ui-patterns/forms/overview#validation-on-submit) for more info.
+
+If the process loads new content, move focus to first focusable element in the newly loaded content. This pattern is also described in the [tree view focus guidelines](/components/tree-view#focus-out).
+
+### Disabling controls during loading
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{gap: 3}}>
+
+<Box flex={1}>
+
+You may disable form controls after the user submits the form to avoid confusion over whether changes that were made during the submission process will be saved.
+
+However, the button that initiates the a loading process may not be disabled.
+
+</Box>
+
+<CustomVideoPlayer
+  width="456"
+  loop
+  autoplay
+  src="https://github.com/primer/design/assets/2313998/cc592284-08d9-4330-8f36-548f27615706"
+/>
+</Box>
+
+### Button loading state
+
+_TODO: Refine this content once the a11y team confirms we interpretted their feedback correctly_
+_TODO: Consider moving most of the button loading state details to the [Button interface guidelines](/components/button)_
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{alignItems: 'start', gap: 3}}>
+
+<Box flex={1}>
+
+When implementing a "loading" button state, don't remove the button from the DOM or pass the `disabled` attribute. Doing so would make it impossible to tab to the button. If the button was just focused and activated, it would reset focus. Resetting focus would disrupt the keyboard navigation flow, and creates a confusing experience for assistive technologies such as screen readers.
+
+</Box>
+
+<img
+  width="456"
+  alt="A button in a loading state with a11y annotations"
+  src="replace-me-with-real-src"
+/>
+</Box>
+
+Once the button is activated (and is in a loading state), it becomes `aria-disabled` and receives `aria-describedby` that indicates that it's processing and may take a while. 
+
+A separate, visually hidden element should be rendered outside of the `<button>` with a message to communicate the loading status. For example, "Saving profile".
+
+The visually hidden status message element receives `aria-live="polite"` and `aria-busy="true"` attributes.
+
+If an error prevents process from being completed, focus should be brought to an `<h2>` (or next relevant heading) of the error banner. 
```

Comment:
I'm going to get more guidance from @ichelsea when she gets back.

I don't want this to block the PR from being reviewed and merged, so I'll comment this out if we're ready to merge before @ichelsea is back.